### PR TITLE
fix(watch): respawn process only after exit

### PR DESCRIPTION
**Pull request by npdev453**
That code cause EADDRINUSE error in 50% of restarts in watch command if port binding going fast. 
Aslo impossible to get around it with graceful shutdown logic.

### CODE REVIEW

File: src/watch/index.ts [hunk 1]
```diff
@@ -98,6 +86,25 @@ export const watchCommand = command({
...
 		});
+	};
+
+	const reRun = debounce(() => {
+		if (
+			runProcess
+			&& (!runProcess.killed && runProcess.exitCode === null)
+		) {
+			runProcess.kill();
+		}
```
Comment (privatenumber) [inline src/watch/index.ts:97]: Can you move this logic down right after the `exit` handler is registered?

Comment (npdev453) [reply src/watch/index.ts:100]: @privatenumber like this?

Comment (ben-xD) [reply src/watch/index.ts:100]: @privatenumber, could you take a look? Fixing this problem would improve developer experience quite a lot

Comment (giva9712) [reply src/watch/index.ts:100]: it is a very useful feature, PTAL @privatenumber

**Comment by privatenumber** [comment]
Maintainer editing is disabled on this pull request, so I can't make changes or run tests.

Closing in favor of https://github.com/esbuild-kit/tsx/pull/298

**Pull request closed without merging**